packages/twenty-front/src/modules/action-menu/actions/record-actions/multiple-records/hooks/useDestroyMultipleRecordsAction.tsx

```diff
@@ -0,0 +1,121 @@
+import { ActionHookWithObjectMetadataItem } from '@/action-menu/actions/types/ActionHook';
+
+import { contextStoreFiltersComponentState } from '@/context-store/states/contextStoreFiltersComponentState';
+import { contextStoreNumberOfSelectedRecordsComponentState } from '@/context-store/states/contextStoreNumberOfSelectedRecordsComponentState';
+import { contextStoreTargetedRecordsRuleComponentState } from '@/context-store/states/contextStoreTargetedRecordsRuleComponentState';
+import { computeContextStoreFilters } from '@/context-store/utils/computeContextStoreFilters';
+import { DEFAULT_QUERY_PAGE_SIZE } from '@/object-record/constants/DefaultQueryPageSize';
+import { DELETE_MAX_COUNT } from '@/object-record/constants/DeleteMaxCount';
+import { RecordGqlOperationFilter } from '@/object-record/graphql/types/RecordGqlOperationFilter';
+import { useDestroyManyRecords } from '@/object-record/hooks/useDestroyManyRecords';
+import { useLazyFetchAllRecords } from '@/object-record/hooks/useLazyFetchAllRecords';
+import { useFilterValueDependencies } from '@/object-record/record-filter/hooks/useFilterValueDependencies';
+import { RecordFilterOperand } from '@/object-record/record-filter/types/RecordFilterOperand';
+import { useRecordTable } from '@/object-record/record-table/hooks/useRecordTable';
+import { ConfirmationModal } from '@/ui/layout/modal/components/ConfirmationModal';
+import { useRecoilComponentValueV2 } from '@/ui/utilities/state/component-state/hooks/useRecoilComponentValueV2';
+import { useCallback, useState } from 'react';
+import { isDefined } from 'twenty-shared';
+
+export const useDestroyMultipleRecordsAction: ActionHookWithObjectMetadataItem =
+  ({ objectMetadataItem }) => {
+    const [isDestroyRecordsModalOpen, setIsDestroyRecordsModalOpen] =
+      useState(false);
+
+    const { resetTableRowSelection } = useRecordTable({
+      recordTableId: objectMetadataItem.namePlural,
+    });
+
+    const { destroyManyRecords } = useDestroyManyRecords({
+      objectNameSingular: objectMetadataItem.nameSingular,
+    });
+
+    const contextStoreNumberOfSelectedRecords = useRecoilComponentValueV2(
+      contextStoreNumberOfSelectedRecordsComponentState,
+    );
+
+    const contextStoreTargetedRecordsRule = useRecoilComponentValueV2(
+      contextStoreTargetedRecordsRuleComponentState,
+    );
+
+    const contextStoreFilters = useRecoilComponentValueV2(
```

Comment:
We can leave this right now but it should become record filters, I'll handle it in my upcoming PRs.

Reply:
okay feel free tagging me as a reviewer !